### PR TITLE
feature: edit icons

### PR DESCRIPTION
**Pull request by csiyang**
# Description

Added a generic component to edit icon blocks. Implemented this for button.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/26244169/todos/4632616824)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Manual testing for changing each of the options

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/api-journeys/src/app/modules/block/icon/icon.resolver.ts [hunk 1]
```diff
@@ -22,11 +22,17 @@ export class IconBlockResolver {
     ])
   )
   @IdAsKey()
+  @UseGuards(
```
Comment (csiyang) [inline apps/api-journeys/src/app/modules/block/icon/icon.resolver.ts:25]: remove

File: apps/api-journeys/src/app/modules/block/icon/icon.resolver.ts [hunk 2]
```diff
@@ -22,11 +22,17 @@ export class IconBlockResolver {
...
+    RoleGuard('input.journeyId', [
+      UserJourneyRole.owner,
+      UserJourneyRole.editor
+    ])
+  )
   async iconBlockCreate(
-    @Args('input') input: IconBlockCreateInput & { __typename }
+    @Args('input') input: IconBlockCreateInput
```
Comment (csiyang) [inline apps/api-journeys/src/app/modules/block/icon/icon.resolver.ts:32]: revert

File: apps/api-journeys/src/app/modules/block/icon/icon.resolver.ts [hunk 3]
```diff
@@ -22,11 +22,17 @@ export class IconBlockResolver {
...
   ): Promise<IconBlock> {
-    input.__typename = 'IconBlock'
     return await this.blockService.save({
+      __typename: 'IconBlock',
```
Comment (csiyang) [inline apps/api-journeys/src/app/modules/block/icon/icon.resolver.ts:35]: revert

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Icon/Icon.tsx [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+      name
+    }
+  }
+`
+
+interface iconProps {
+  iconBlock: TreeBlock<IconFields>
+}
+
+// TODO: bug - for some reason, sometimes can't switch to certain icons, don't know how to recreate
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Icon/Icon.tsx:126]: remove

File: apps/journeys-admin/src/libs/apolloClient/cache.ts [hunk 5]
```diff
@@ -9,6 +9,9 @@ export const cache = (): InMemoryCache =>
      initializing InMemoryCache.
    */
     typePolicies: {
+      Icon: {
```
Comment (csiyang) [inline apps/journeys-admin/src/libs/apolloClient/cache.ts:12]: Do I still need this?

File: libs/journeys/ui/src/components/index.ts [hunk 6]
```diff
@@ -11,3 +11,4 @@ export { Step } from './Step'
 export { Typography } from './Typography'
 export { Video } from './Video'
 export { VideoTrigger } from './Video/VideoTrigger'
+export { Icon } from './Icon'
```
Comment (csiyang) [inline libs/journeys/ui/src/components/index.ts:14]: revert

File: libs/journeys/ui/src/index.ts [hunk 7]
```diff
@@ -12,7 +12,8 @@ export {
   Step,
   Typography,
   Video,
-  VideoTrigger
+  VideoTrigger,
+  Icon
```
Comment (csiyang) [inline libs/journeys/ui/src/index.ts:16]: revert

File: package-lock.json [hunk 8]
```diff
@@ -25,6 +25,7 @@
         "@nestjs/core": "^8.2.3",
         "@nestjs/graphql": "9.1.2",
         "@nestjs/platform-express": "^8.2.3",
+        "@next/swc-linux-arm64-gnu": "^12.0.9",
```
Comment (csiyang) [inline package-lock.json:28]: remove

File: package.json [hunk 9]
```diff
@@ -47,6 +47,7 @@
     "@nestjs/core": "^8.2.3",
     "@nestjs/graphql": "9.1.2",
     "@nestjs/platform-express": "^8.2.3",
+    "@next/swc-linux-arm64-gnu": "^12.0.9",
```
Comment (csiyang) [inline package.json:50]: remove

File: apps/journeys-admin/src/libs/apolloClient/cache.ts (hunk 5 above)
Comment (csiyang) [reply apps/journeys-admin/src/libs/apolloClient/cache.ts:12]: Nope.

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Icon/Icon.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,172 @@
+import { render, fireEvent, waitFor } from '@testing-library/react'
+import { MockedProvider } from '@apollo/client/testing'
+import { EditorProvider, TreeBlock } from '@core/journeys/ui'
+import { IconName } from '../../../../../../__generated__/globalTypes'
+import { IconFields } from '../../../../../../__generated__/IconFields'
+import { ICON_BLOCK_NAME_UPDATE } from './Icon'
+import { Icon } from '.'
+
+describe('Icon', () => {
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Icon/Icon.spec.tsx:9]: Nice tests :) Good naming and thorough !

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Icon/SizeToggleGroup/SizeToggleGroup.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,33 @@
+import { Story, Meta } from '@storybook/react'
+import { MockedProvider } from '@apollo/client/testing'
+import { TreeBlock } from '@core/journeys/ui'
+import { simpleComponentConfig } from '../../../../../../libs/storybook'
+import { IconFields } from '../../../../../../../__generated__/IconFields'
+import { IconName } from '../../../../../../../__generated__/globalTypes'
+import { SizeToggleGroup } from '.'
+
+const SizeToggleGroupStory = {
+  ...simpleComponentConfig,
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Icon/SizeToggleGroup/SizeToggleGroup.stories.tsx:10]: Good thinking - using simpleComponentConfig for these smaller toggle groups and journeyAdminConfig for drawer.

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Button/Button.tsx [hunk 12]
```diff
@@ -101,18 +102,38 @@ export function Button({
         id={`${id}-button-leading-icon`}
         icon={<InfoOutlinedIcon />}
         name="Leading Icon"
-        value={startIcon?.iconName ?? 'None'}
+        value={
+          icons.find(({ value }) => value === startIcon?.iconName)?.label ??
```
Comment (storyworks) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Button/Button.tsx:106]: Can we not keep this as `startIcon.iconName ?? 'None'`, similar for endIcon

File: libs/journeys/ui/src/components/Icon/Icon.tsx [hunk 13]
```diff
@@ -58,7 +58,7 @@ export function Icon({
   ) : (
     createElement(icons[iconName], {
       color: iconColor ?? undefined,
-      sx: { fontSize }
+      sx: { width: fontSize, height: fontSize }
```
Comment (storyworks) [inline libs/journeys/ui/src/components/Icon/Icon.tsx:61]: Why did you change this one? Material recommends use to use fontSize to determine the width / height of icons https://mui.com/components/icons/#size

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Icon/SizeToggleGroup/SizeToggleGroup.stories.tsx (hunk 11 above)
Comment (csiyang) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Icon/SizeToggleGroup/SizeToggleGroup.stories.tsx:10]: 👌

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Icon/Icon.spec.tsx (hunk 10 above)
Comment (csiyang) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/Icon/Icon.spec.tsx:9]: 🙌

File: libs/journeys/ui/src/components/Icon/Icon.tsx (hunk 13 above)
Comment (csiyang) [reply libs/journeys/ui/src/components/Icon/Icon.tsx:61]: Removed the ability to change size on icon as MuiButton overrides the size set for icon

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Button/Button.tsx (hunk 12 above)
Comment (csiyang) [reply apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Button/Button.tsx:106]: We need this change so that the naming is consistent between the icon selector and the attribute button. Also needed for internationalisation.

**Review by storyworks — APPROVED** [review]
